### PR TITLE
fix force delete pod when node is offline

### PR DESCRIPTION
**Pull request by wackxu**
Signed-off-by: wackxu <xushiwei5@huawei.com>

**What type of PR is this?**

/kind bug
**What this PR does / why we need it**:

**Which issue(s) this PR fixes**:

Fixes # 

when node is offline and we force delete the pod in the node, and then the edge node is online,
the deleted pod container is always in the edge node unless we restart the edgecore.

```
	// When the edge node is offline and the pod in the node is deleted forcefully,
	// and then we make the node online, We do not have the pod full information
	// because the pod is deleted from the kube apiServer, then the syncController
	// will send a message with the pod name, namespace and UID, so we can not filter
	// pod according to the node name, if the pod exist in the podManager, it indicates
	// that the pod is belong to this edge node, so we can delete pod from the edged,
	// otherwise the pod is not belong to the node, we just return.
```

The releate E2E will be added in anoter pr

**Special notes for your reviewer**:

**Does this PR introduce a user-facing change?**:

```release-note

```

### CODE REVIEW

File: edge/pkg/edged/edged.go [hunk 1]
```diff
@@ -1310,16 +1310,26 @@ func (e *edged) handlePod(op string, content []byte) (err error) {
...
+	// that the pod is belong to this edge node, so we can delete pod from the edged,
+	// otherwise the pod is not belong to the node, we just return.
+	if op == model.DeleteOperation {
+		if delPod, ok := e.podManager.GetPodByName(pod.Namespace, pod.Name); ok {
+			e.deletePod(delPod)
+		}
+		return nil
+	}
+
 	if filterPodByNodeName(&pod, e.nodeName) {
```
Comment (gy95) [inline edge/pkg/edged/edged.go:1327]: you mean in this circumstance, here pod doesn't contain node info, so will not trigger a delete operation?

Comment (wackxu) [reply edge/pkg/edged/edged.go:1327]: yes, in this case, 
```
	if apierrors.IsNotFound(err) {
		// trigger the delete event
		klog.V(4).Infof("%s: %s has been deleted in K8s, send the delete event to edge in sync loop", resourceType, sync.Spec.ObjectName)
		newObject := &unstructured.Unstructured{}
		newObject.SetNamespace(sync.Namespace)
		newObject.SetName(sync.Spec.ObjectName)
		newObject.SetUID(types.UID(getObjectUID(sync.Name)))
		if msg := buildEdgeControllerMessage(nodeName, sync.Namespace, resourceType, sync.Spec.ObjectName, model.DeleteOperation, newObject); msg != nil {
			beehiveContext.Send(commonconst.DefaultContextSendModuleName, *msg)
		} else {
			if err := sctl.crdclient.ReliablesyncsV1alpha1().ObjectSyncs(sync.Namespace).Delete(context.Background(), sync.Name, *metav1.NewDeleteOptions(0)); err != nil {
				klog.Errorf("Failed to delete objectsync %s: %v", sync.Name, err)
			}
		}
		return
	}
```

**Comment by wackxu** [comment]
Steps to reproduce the problem

1、create pod in the edge node and wait the pod is running
2、Simulation node is offline, we can use `chaosmesh` or iptables
3、when the node is offline, force delete the pod
`kubectl delete pod xx --force --grace-period=0`
4、make the node online
5、check the container of the pod is running and is not deleted.

File: edge/pkg/edged/edged.go (hunk 1 above)
Comment (wackxu) [reply edge/pkg/edged/edged.go:1327]: for most case we can get the node info from the pod

**Comment by Shelley-BaoYue** [comment]
In other normal deletion scenarios，will this modification cause some problem？

**Comment by wackxu** [comment]
> the pod is not belong to the node, we just return.

No, if the pod is belong to this node, we could find it from the pod manager, and the pod will be deleted from edged.

If  the pod is not belong to the node,  we could not find the info from the pod manager, so the delete message will not be processed and return directlly.

File: edge/pkg/edged/edged.go (hunk 1 above)
Comment (gy95) [reply edge/pkg/edged/edged.go:1327]: /lgtm

**Review by fisherxu — COMMENTED** [review]
/lgtm
And if setting the nodename for pod in synccontroller, the issue can also be fixed? :) If fixed in edged, do the new kubelet still need this fixing?

**Comment by Shelley-BaoYue** [comment]
> /lgtm And if setting the nodename for pod in synccontroller, the issue can also be fixed? :) If fixed in edged, do the new kubelet still need this fixing?

The new Kubelet also needs. I will rebase and add this fixing.

**Comment by wackxu** [comment]
> /lgtm And if setting the nodename for pod in synccontroller, the issue can also be fixed? :) If fixed in edged, do the new kubelet still need this fixing?

Set nodeName in the synccontroller can also solve the problem. but we need to sense whether the  kind is a pod or not, and then set the name of the node

**Comment by fisherxu** [comment]
PR need rebase :)

**Comment by wackxu** [comment]
> PR need rebase :)

done

**Review by fisherxu — COMMENTED** [review]
/lgtm
/approve

**Comment by fisherxu** [comment]
Please cherry-pick the PR to release1.9~1.11 :)

**Pull request merged**